File: app/src/main/java/org/sabda/gpt/Tentang.java
package org.sabda.gpt;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.ceylonlabs.imageviewpopup.BuildConfig;

public class Tentang extends AppCompatActivity implements NetworkUtil.NetworkChangeCallback {

    ImageView back;
    TextView version, sabdaLink, kotakSaranLink;
    boolean isConnected = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tentang);

        isConnected = NetworkUtil.isNetworkAvailable(this);
        back = findViewById(R.id.back);
        version = findViewById(R.id.version);
        sabdaLink = findViewById(R.id.sabdaLink);
        kotakSaranLink = findViewById(R.id.kotakSaranLink);

        updateConnectionStatus(isConnected);

        NetworkUtil.registerNetworkChangeReceiver(this, this::updateConnectionStatus);

        back.setOnClickListener(v -> finish());

        version.setText("Versi: 2.2");

        sabdaLink.setOnClickListener(v -> handleClick("https://sabda.app"));

        kotakSaranLink.setOnClickListener(v -> handleClick("mailto:apps@sabda.org"));

    }

    private void updateConnectionStatus(boolean isConnected){
        this.isConnected = isConnected;
        sabdaLink.setEnabled(isConnected);
        kotakSaranLink.setEnabled(isConnected);

        if (!isConnected){
            ToastUtil.showToast(this);
        }
    }

    private void handleClick(String url){
        if (isConnected){
            NetworkUtil.openUrl(this, url);
        } else {
            ToastUtil.showToast(this);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        NetworkUtil.unregisterNetworkChangeReceiver(this);
    }

    @Override
    public void onNetworkChange(boolean isConnected) {
        updateConnectionStatus(isConnected);
    }
}

File: app/src/main/java/org/sabda/gpt/Tentang.kt
package org.sabda.gpt

import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import org.sabda.gpt.utility.NetworkUtil
import org.sabda.gpt.utility.NetworkUtil.NetworkChangeCallback
import org.sabda.gpt.utility.ToastUtil

class Tentang : AppCompatActivity(), NetworkChangeCallback {
    private var back: ImageView? = null
    var version: TextView? = null
    private var sabdaLink: TextView? = null
    private var kotakSaranLink: TextView? = null
    private var isConnected: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tentang)

        isConnected = NetworkUtil.isNetworkAvailable(this)
        back = findViewById(R.id.back)
        version = findViewById(R.id.version)
        sabdaLink = findViewById(R.id.sabdaLink)
        kotakSaranLink = findViewById(R.id.kotakSaranLink)

        updateConnectionStatus(isConnected)

        NetworkUtil.registerNetworkChangeReceiver(
            this
        ) { isConnected: Boolean -> this.updateConnectionStatus(isConnected) }

        back!!.setOnClickListener { finish() }

        version!!.text = "Versi: 2.2"

        sabdaLink!!.setOnClickListener { handleClick("https://sabda.app") }

        kotakSaranLink!!.setOnClickListener { handleClick("mailto:apps@sabda.org") }
    }

    private fun updateConnectionStatus(isConnected: Boolean) {
        this.isConnected = isConnected
        sabdaLink!!.isEnabled = isConnected
        kotakSaranLink!!.isEnabled = isConnected

        if (!isConnected) {
            ToastUtil.showToast(this)
        }
    }

    private fun handleClick(url: String) {
        if (isConnected) {
            NetworkUtil.openUrl(this, url)
        } else {
            ToastUtil.showToast(this)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        NetworkUtil.unregisterNetworkChangeReceiver(this)
    }

    override fun onNetworkChange(isConnected: Boolean) {
        updateConnectionStatus(isConnected)
    }
}